packages/features/bookings/lib/handleNewBooking/createBooking.ts

```diff
@@ -0,0 +1,267 @@
+  }
+}
+
+export async function createBooking({
+  originalRescheduledBooking,
+  evt,
+  eventTypeId,
+  eventTypeSlug,
+  reqBodyUser,
+  reqBodyMetadata,
+  reqBodyRecurringEventId,
+  uid,
+  responses,
+  isConfirmedByDefault,
+  smsReminderNumber,
+  organizerUser,
+  rescheduleReason,
+  eventType,
+  bookerEmail,
+  paymentAppData,
+  changedOrganizer,
+}: CreateBookingParams) {
+  updateEventDetails(evt, originalRescheduledBooking, changedOrganizer);
+
+  const newBookingData = buildNewBookingData({
+    uid,
+    evt,
+    responses,
+    isConfirmedByDefault,
+    reqBodyMetadata,
+    smsReminderNumber,
+    eventTypeSlug,
+    organizerUser,
+    reqBodyRecurringEventId,
+    originalRescheduledBooking,
+    bookerEmail,
+    rescheduleReason,
+    eventType,
+    eventTypeId,
+    reqBodyUser,
+  });
+
+  return await saveBooking(newBookingData, originalRescheduledBooking, paymentAppData, organizerUser);
+}
+
+async function saveBooking(
+  newBookingData: Prisma.BookingCreateInput,
+  originalRescheduledBooking: OriginalRescheduledBooking,
+  paymentAppData: PaymentAppData,
+  organizerUser: CreateBookingParams["organizerUser"]
+) {
+  const createBookingObj = {
+    include: {
+      user: {
+        select: { email: true, name: true, timeZone: true, username: true },
+      },
+      attendees: true,
+      payment: true,
+      references: true,
+    },
+    data: newBookingData,
+  };
+
+  if (originalRescheduledBooking?.paid && originalRescheduledBooking?.payment) {
+    const bookingPayment = originalRescheduledBooking.payment.find((payment) => payment.success);
+    if (bookingPayment) {
+      createBookingObj.data.payment = { connect: { id: bookingPayment.id } };
+    }
+  }
+
+  if (typeof paymentAppData.price === "number" && paymentAppData.price > 0) {
+    await prisma.credential.findFirstOrThrow({
+      where: {
+        appId: paymentAppData.appId,
+        ...(paymentAppData.credentialId ? { id: paymentAppData.credentialId } : { userId: organizerUser.id }),
+      },
+      select: { id: true },
+    });
+  }
+
+  return prisma.booking.create(createBookingObj);
+}
+
+function getEventTypeRel(eventTypeId: EventTypeId) {
+  return eventTypeId ? { connect: { id: eventTypeId } } : {};
+}
+
+function getAttendeesData(evt: Pick<CalendarEvent, "attendees" | "team">) {
+  //if attendee is team member, it should fetch their locale not booker's locale
+  //perhaps make email fetch request to see if his locale is stored, else
+  const attendees = evt.attendees.map((attendee) => ({
+    name: attendee.name,
+    email: attendee.email,
+    timeZone: attendee.timeZone,
+    locale: attendee.language.locale,
+  }));
+
+  if (evt.team?.members) {
+    attendees.push(
+      ...evt.team.members.map((member) => ({
+        email: member.email,
+        name: member.name,
+        timeZone: member.timeZone,
+        locale: member.language.locale,
+      }))
+    );
+  }
```

Comment:
Followup: Simplification
```suggestion
 const teamMembers = event.team?.members ?? []
 return evt.attendees.concat(teamMembers).map(()=>{
 name: attendee.name,
    email: attendee.email,
    timeZone: attendee.timeZone,
    locale: attendee.language.locale,
 })
```